### PR TITLE
Include uiSchema in Template forms

### PR DESCRIPTION
**Pull request by OscarDHdz**
### Problem Statement
Unable to add textarea fields in a template. 

### Solution  
Allowing templates to make use of the [react-jsonschema-form uiSchema](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/uiSchema/) to modify how the form is rendered. (_Not only textarea_ 😎)

### Changes
* Adding `uiSchema` prop to `MultistepJsonForm`
* Enabling `uiSchema` object in Templates **YAML** files 

> Template Page. 

Screen Shot 2021-02-18 at 8 56 36. 

> Template YML.  

Screen Shot 2021-02-18 at 8 49 25

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4584-pdqnjq6e7pxtsmeuekrerwzar1hvqcsg.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602ed26afcb22db8955a19cd

**Review by Rugvip — COMMENTED** [review]
Thanks!

We've actually just been working on this today for the next steps of the scaffolder.

So far we're thinking we'll be merging the schema and the uiSchema together, since as far as we could find it's only done that way to have a clearer separation for users that want to for example being in their JSON Schema from a remove source. More info on that here https://github.com/rjsf-team/react-jsonschema-form/issues/701

Basically doing something like this:

```yaml
parameters:
  required:
    - name
  properties:
    name:
      title: Name
      type: string
      description: Unique name of the component
      ui:widget: textarea
      ui:options:
        rows: 5
```

Do you have any opinions on that approach, and would it be enough for your needs?

**Comment by OscarDHdz** [comment]
@Rugvip Absolutely! that would be enough for our needs 👌.
Just to double check... you guys will be taking care of that, right??

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4584-pdqnjq6e7pxtsmeuekrerwzar1hvqcsg.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602ed26afcb22db8955a19cd

**Comment by OscarDHdz** [comment]
> We've actually just been working on this today for the next steps of the scaffolder.

Just one doubt... are you guys doing that within backstage?, or are you guys gonna wait for rjsf-team/react-jsonschema-form#701 to be resolved? 

If there's something I can help with, let me know 😉, or else I think we can close this PR 😎

**Comment by freben** [comment]
Yep we will close for now but can hold on to the inspiration. We are wiring together something very similar to this within Backstage as we speak :) Great minds, and all that
